src/BinSkim.Driver/MultithreadedAnalyzeCommand.cs

```diff
@@ -71,13 +71,47 @@
 
         internal static string ReturnCommonPathRootFromTargetSpecifiersIfOneExists(IEnumerable<string> targetFileSpecifiers)
         {
+            var fileSpecifierDirectories = new SortedSet<string>(targetFileSpecifiers.Select(s => (Path.GetDirectoryName(Path.GetFullPath(s)) + @"\")),
+                                                                 StringComparer.OrdinalIgnoreCase);
 
-            var fileSpecifierDirectories = new HashSet<string>(targetFileSpecifiers.Select(s => Path.GetDirectoryName(Path.GetFullPath(s)) + @"\"),
-                                                               StringComparer.OrdinalIgnoreCase);
+            string smallestPath = fileSpecifierDirectories.First();
+            fileSpecifierDirectories.Remove(smallestPath);
 
-            return fileSpecifierDirectories.Count == 1
-                ? fileSpecifierDirectories.First()
-                : string.Empty;
+            for (int i = 0; i < smallestPath.Length; i++)
+            {
+                // We don't need to iterate all characters of the smallestPath if we don't have anything
+                // to compare against.
```

Comment:
>ToLowerInvariant [](http://example.com/codeflow?start=29&length=16)

are you subject to Turkish i issues here?

Reply:
Since I'm calling chart.ToLowerInvariant, this wont face the `Turkish i issue`. Thanks for pointing that out, I didn't know about that issue.